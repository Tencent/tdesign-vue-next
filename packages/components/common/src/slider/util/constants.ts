import type { InjectionKey } from '@td/adapter-vue';

type SliderPropsInjectKey = InjectionKey<{
  max: number;
  min: number;
  step: number;
  dragging: boolean;
  toggleDragging: (val: boolean) => void;
  precision: number;
  disabled: boolean;
  resetSize: () => void;
  sliderSize: number;
}>;
export const sliderPropsInjectKey: SliderPropsInjectKey = Symbol('sliderProps');
