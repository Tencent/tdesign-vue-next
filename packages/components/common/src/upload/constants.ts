import type { PropType } from '@td/adapter-vue';
import type { CommonDisplayFileProps } from './interface';

export const commonProps = {
  accept: String,
  files: Array as PropType<CommonDisplayFileProps['files']>,
  toUploadFiles: Array as PropType<CommonDisplayFileProps['toUploadFiles']>,
  displayFiles: Array as PropType<CommonDisplayFileProps['displayFiles']>,
  theme: String as PropType<CommonDisplayFileProps['theme']>,
  abridgeName: Array as PropType<CommonDisplayFileProps['abridgeName']>,
  placeholder: String,
  classPrefix: String,
  tips: [String, Function],
  status: String as PropType<CommonDisplayFileProps['status']>,
  locale: Object as PropType<CommonDisplayFileProps['locale']>,
  sizeOverLimitMessage: String,
  autoUpload: Boolean,
  uploadPastedFiles: Boolean,
  onPasteFileChange: Function as PropType<CommonDisplayFileProps['onPasteFileChange']>,
  disabled: Boolean,
  uploading: Boolean,
  tipsClasses: [String, Array, Object] as PropType<CommonDisplayFileProps['tipsClasses']>,
  errorClasses: [String, Array, Object] as PropType<CommonDisplayFileProps['errorClasses']>,
  placeholderClass: [String, Array, Object] as PropType<CommonDisplayFileProps['placeholderClass']>,
  showUploadProgress: Boolean,
  default: Function as PropType<CommonDisplayFileProps['default']>,
  fileListDisplay: Function as PropType<CommonDisplayFileProps['fileListDisplay']>,
  onRemove: Function as PropType<CommonDisplayFileProps['onRemove']>,
  imageViewerProps: Object as PropType<CommonDisplayFileProps['imageViewerProps']>,
};
