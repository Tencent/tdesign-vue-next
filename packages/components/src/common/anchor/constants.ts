import type { InjectionKey } from '@td/adapter-vue';

export const AnchorInjectionKey: InjectionKey<{
  registerLink: (link: string) => void;
  unregisterLink: (link: string) => void;
  handleScrollTo: (link: string) => void;
  handleLinkClick: (link: { href: string; title: string; e: MouseEvent }) => void;
  active: string;
}> = Symbol('AnchorInjectionProvide');
