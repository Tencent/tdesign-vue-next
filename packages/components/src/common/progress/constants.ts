export const PRO_THEME = {
  LINE: 'line',
  PLUMP: 'plump',
  CIRCLE: 'circle',
};
export const CIRCLE_SIZE = {
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
};
export const CIRCLE_SIZE_PX = {
  SMALL: 72,
  MEDIUM: 112,
  LARGE: 160,
};
export const CIRCLE_FONT_SIZE_RATIO = {
  SMALL: 14 / CIRCLE_SIZE_PX.SMALL,
  MEDIUM: 20 / CIRCLE_SIZE_PX.MEDIUM,
  LARGE: 36 / CIRCLE_SIZE_PX.LARGE,
};
export const STATUS_TEXT = ['success', 'error', 'warning', 'active', 'normal'];
export const STATUS_ICON = ['success', 'error', 'warning'];

export default {
  PRO_THEME,
  CIRCLE_SIZE,
  CIRCLE_SIZE_PX,
  STATUS_TEXT,
  STATUS_ICON,
};
