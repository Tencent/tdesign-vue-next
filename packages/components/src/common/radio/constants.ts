import type { InjectionKey } from '@td/adapter-vue';
import type { RadioValue } from '@td/intel/components/radio/type';

export const RadioGroupInjectionKey: InjectionKey<{
  name: string;
  disabled: boolean;
  value: RadioValue;
  allowUncheck: boolean;
  setValue: (value: RadioValue, context: { e: Event }) => void;
}> = Symbol('RadioGroupProvide');

export const RadioButtonInjectionKey: InjectionKey<{}> = Symbol('RadioButtonProvide');
