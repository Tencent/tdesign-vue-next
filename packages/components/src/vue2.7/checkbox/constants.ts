import type { ComputedRef, InjectionKey }  from '@td/adapter-vue'
import type { TdCheckboxProps } from '@td/intel/components/checkbox/type';

export interface CheckboxGroupInjectData {
  handleCheckboxChange: (data: { checked: boolean; e: Event; option: TdCheckboxProps }) => void;
  onCheckedChange: (p: { checked: boolean; checkAll: boolean; e: Event; option: TdCheckboxProps }) => void;
}

export const CheckboxGroupInjectionKey: InjectionKey<ComputedRef<CheckboxGroupInjectData>> = Symbol('CheckboxGroupProvide');
