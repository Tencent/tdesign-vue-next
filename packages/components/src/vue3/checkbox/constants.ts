import type { ComputedRef, InjectionKey } from '@td/adapter-vue';
import type { TdCheckboxGroupProps, TdCheckboxProps } from '@td/intel/components/checkbox/type';

export interface CheckboxGroupInjectData {
  name?: string;
  isCheckAll: boolean;
  maxExceeded: boolean;
  disabled: boolean;
  indeterminate: boolean;
  checkedValues: TdCheckboxGroupProps['value'];
  handleCheckboxChange: (data: { checked: boolean; e: Event; option: TdCheckboxProps }) => void;
  onCheckedChange: (p: { checked: boolean; checkAll: boolean; e: Event; option: TdCheckboxProps }) => void;
}

export const CheckboxGroupInjectionKey: InjectionKey<ComputedRef<CheckboxGroupInjectData>> =
  Symbol('CheckboxGroupProvide');
