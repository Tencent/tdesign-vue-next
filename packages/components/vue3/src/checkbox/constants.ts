import { ComputedRef, InjectionKey } from 'vue';
import { TdCheckboxGroupProps, TdCheckboxProps } from '@td/intel/../../vue3/src/checkbox/type';

export interface CheckboxGroupInjectData {
  name?: string;
  isCheckAll: boolean;
  maxExceeded: boolean;
  disabled: boolean;
  indeterminate: boolean;
  checkedValues: TdCheckboxGroupProps['value'];
  handleCheckboxChange: (data: { checked: boolean; e: Event; option: TdCheckboxProps }) => void;
  onCheckedChange: (p: { checked: boolean; checkAll: boolean; e: Event; option: TdCheckboxProps }) => void;
}

export const CheckboxGroupInjectionKey: InjectionKey<ComputedRef<CheckboxGroupInjectData>> =
  Symbol('CheckboxGroupProvide');
