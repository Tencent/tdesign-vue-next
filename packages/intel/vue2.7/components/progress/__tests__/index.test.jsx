import { mount } from '@vue/test-utils';
import { Progress } from '@/src/progress/index.ts';
import { PRO_THEME, CIRCLE_SIZE } from '@/src/progress/constants.ts';

const STATUS_TEXT = ['success', 'error', 'warning', 'active'];

describe('Progress', () => {
  it('pure progress contains one class', () => {
    const wrapper = mount(Progress);
    const classes = wrapper.classes();
    expect(classes).toContain('t-progress');
    expect(wrapper.element).toMatchSnapshot();
  });

  describe(':props', () => {
    describe(':theme', () => {
      Object.values(PRO_THEME).forEach((theme) => {
        it(`:theme:${theme}`, () => {
          const wrapper = mount({
            render() {
              return <Progress theme={theme}></Progress>;
            },
          });
          expect(wrapper.findComponent(Progress).vm.theme).toBe(theme);
          expect(wrapper.element).toMatchSnapshot();
        });
      });
      it(':theme line and plump progress have t-progress__inner class', () => {
        const lineWrapper = mount({
          render() {
            return <Progress theme={'line'}></Progress>;
          },
        });
        const lineClass = lineWrapper.element.querySelector('.t-progress__inner');

        expect(lineClass !== undefined && lineClass !== null).toBe(true);
        expect(lineWrapper.element).toMatchSnapshot();

        // < 10% plump
        const plumpWrapper = mount({
          render() {
            return <Progress theme={'plump'}></Progress>;
          },
        });
        const plumpClass = plumpWrapper.element.querySelector('.t-progress__inner');

        expect(plumpClass !== undefined && plumpClass !== null).toBe(true);
        expect(plumpWrapper.element).toMatchSnapshot();
        expect(plumpWrapper.element.querySelector('.t-progress__info').innerHTML).toBe('0%');

        // =10% plump
        const plump10Wrapper = mount({
          render() {
            return <Progress theme={'plump'} percentage={10}></Progress>;
          },
        });
        const plump10Class = plump10Wrapper.element.querySelector('.t-progress__inner');

        expect(plump10Class !== undefined && plump10Class !== null).toBe(true);
        expect(plump10Wrapper.element).toMatchSnapshot();
        expect(plump10Wrapper.element.querySelector('.t-progress__info').innerHTML).toBe('10%');

        // >10% plump
        const plump11Wrapper = mount({
          render() {
            return <Progress theme={'plump'} percentage={11}></Progress>;
          },
        });
        const plump11Class = plump11Wrapper.element.querySelector('.t-progress__inner');

        expect(plump11Class !== undefined && plump11Class !== null).toBe(true);
        expect(plump11Wrapper.element).toMatchSnapshot();
        expect(plump11Class.querySelector('.t-progress__info').innerHTML).toBe('11%');
      });

      it(':theme circle progress has t-progress__circle-outer and t-progress__circle-inner class', () => {
        const wrapper = mount({
          render() {
            return <Progress theme={'circle'} percentage={'10'}></Progress>;
          },
        });
        const outerClass = wrapper.element.querySelector('.t-progress__circle-outer');
        const innerClass = wrapper.element.querySelector('.t-progress__circle-inner');
        expect(outerClass !== undefined && outerClass !== null).toBe(true);
        expect(innerClass !== undefined && innerClass !== null).toBe(true);
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    describe(':percentage', () => {
      it(':percentage is between 0 to 100, equal to 50', () => {
        const wrapper = mount({
          render() {
            return <Progress percentage={50}></Progress>;
          },
        });
        expect(wrapper.findComponent(Progress).vm.percentage).toBe(50);
        expect(wrapper.element).toMatchSnapshot();
      });
      it(':percentage is less than 0, equal to -10', () => {
        const wrapper = mount({
          render() {
            return <Progress percentage={-10}></Progress>;
          },
        });
        expect(wrapper.findComponent(Progress).vm.percentage).toBe(-10);
        expect(wrapper.element).toMatchSnapshot();
      });
      it(':percentage is great equal than 100, equal to 200', () => {
        const wrapper = mount({
          render() {
            return <Progress percentage={200}></Progress>;
          },
        });
        expect(wrapper.findComponent(Progress).vm.percentage).toBe(200);
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    // label相关单测
    describe(':label', () => {
      it(':label is String, equal to "custom label test"', () => {
        const wrapper = mount({
          render() {
            return <Progress label={'custom label test'}></Progress>;
          },
        });

        expect(wrapper.findComponent(Progress).vm.label).toBe('custom label test');
        expect(wrapper.element).toMatchSnapshot();
      });
      it(':label is boolean, equal to true', () => {
        const wrapper = mount({
          render() {
            return <Progress label={true}></Progress>;
          },
        });

        expect(wrapper.findComponent(Progress).vm.label).toBe(true);
        expect(wrapper.element).toMatchSnapshot();
      });
      it(':label is boolean, equal to false', () => {
        const wrapper = mount({
          render() {
            return <Progress label={false}></Progress>;
          },
        });

        expect(wrapper.findComponent(Progress).vm.label).toBe(false);
        expect(wrapper.element).toMatchSnapshot();
      });
      it(':label is TNode, equal to () => "tnode"', () => {
        const wrapper = mount({
          render() {
            return <Progress label={() => 'tnode'}></Progress>;
          },
        });

        const el = wrapper.element.querySelector('.t-progress__info');
        expect(el.textContent).toBe('tnode');
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    describe(':status', () => {
      STATUS_TEXT.forEach((status) => {
        it(`:status is ${status}`, () => {
          const wrapper = mount({
            render() {
              return <Progress status={status}></Progress>;
            },
          });
          expect(wrapper.findComponent(Progress).vm.status).toBe(status);
          expect(wrapper.element).toMatchSnapshot();
        });

        it(`:status is ${status} and percentage is 100`, () => {
          const wrapper = mount({
            render() {
              return <Progress status={status} percentage={100}></Progress>;
            },
          });
          expect(wrapper.findComponent(Progress).vm.percentage).toBe(100);
          expect(wrapper.findComponent(Progress).vm.status).toBe(status);
          expect(wrapper.element).toMatchSnapshot();
        });
      });
    });

    // color相关单测
    describe(':color', () => {
      it(':color is String, equal to "#aaa"', () => {
        const wrapper = mount({
          render() {
            return <Progress color={'#aaa'}></Progress>;
          },
        });
        expect(wrapper.findComponent(Progress).vm.color).toBe('#aaa');
        expect(wrapper.element).toMatchSnapshot();
      });
      it(':color is Object, equal to { "0%": "#f00", "100%": "#0ff" }', () => {
        const wrapper = mount({
          render() {
            return <Progress color={{ '0%': '#f00', '100%': '#0ff' }}></Progress>;
          },
        });
        expect(wrapper.element).toMatchSnapshot();
      });
      it(':color is Array, equal to ["#f00", "#0ff", "#f0f"]', () => {
        const wrapper = mount({
          render() {
            return <Progress color={['#f00', '#0ff', '#f0f']}></Progress>;
          },
        });
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    describe(':trackColor', () => {
      it(':trackColor is "#aaa"', () => {
        const wrapper = mount({
          render() {
            return <Progress trackColor={'#aaa'}></Progress>;
          },
        });
        expect(wrapper.findComponent(Progress).vm.trackColor).toBe('#aaa');
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    // strokeWidth 相关单测
    describe(':strokeWidth', () => {
      it(':strokeWidth is String, equal to "50px"', () => {
        const wrapper = mount({
          render() {
            return <Progress strokeWidth={'50px'}></Progress>;
          },
        });
        expect(wrapper.findComponent(Progress).vm.strokeWidth).toBe('50px');
        expect(wrapper.element).toMatchSnapshot();
      });
      it(':strokeWidth is Number, equal to 50', () => {
        const wrapper = mount({
          render() {
            return <Progress strokeWidth={50}></Progress>;
          },
        });
        expect(wrapper.findComponent(Progress).vm.strokeWidth).toBe(50);
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    // size相关单测
    describe(':size', () => {
      Object.values(CIRCLE_SIZE).forEach((size) => {
        it(`:size is String, equal to "${size}"`, () => {
          const wrapper = mount({
            render() {
              return <Progress size={size}></Progress>;
            },
          });
          expect(wrapper.findComponent(Progress).vm.size).toBe(size);
          expect(wrapper.element).toMatchSnapshot();
        });
      });
      it(':size is Number, equal to 50', () => {
        const wrapper = mount({
          render() {
            return <Progress size={50}></Progress>;
          },
        });
        expect(wrapper.findComponent(Progress).vm.size).toBe(50);
        expect(wrapper.element).toMatchSnapshot();
      });
    });
  });
});
