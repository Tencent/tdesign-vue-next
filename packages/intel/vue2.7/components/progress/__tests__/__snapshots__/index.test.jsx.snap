// Vitest Snapshot v1

exports[`Progress > :props > :color > :color is Array, equal to ["#f00", "#0ff", "#f0f"] 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :color > :color is Object, equal to { "0%": "#f00", "100%": "#0ff" } 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :color > :color is String, equal to "#aaa" 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%; background: rgb(170, 170, 170);"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :label > :label is String, equal to "custom label test" 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      custom label test
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :label > :label is TNode, equal to () => "tnode" 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      tnode
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :label > :label is boolean, equal to false 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    />
  </div>
</div>
`;

exports[`Progress > :props > :label > :label is boolean, equal to true 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :percentage > :percentage is between 0 to 100, equal to 50 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 50%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      50%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :percentage > :percentage is great equal than 100, equal to 200 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--success"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 200%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      200%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :percentage > :percentage is less than 0, equal to -10 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: -10%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      -10%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :size > :size is Number, equal to 50 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :size > :size is String, equal to "large" 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :size > :size is String, equal to "medium" 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :size > :size is String, equal to "small" 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :status > :status is active 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--active"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :status > :status is active and percentage is 100 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--active"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 100%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      100%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :status > :status is error 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--error"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      <svg
        class="t-icon t-icon-close-circle-filled t-progress__icon"
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12 23a11 11 0 100-22 11 11 0 000 22zM8.82 7.4L12 10.6l3.18-3.19 1.42 1.42L13.4 12l3.19 3.18-1.42 1.42L12 13.4 8.82 16.6 7.4 15.18 10.6 12 7.4 8.82 8.82 7.4z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :status > :status is error and percentage is 100 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--error"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 100%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      <svg
        class="t-icon t-icon-close-circle-filled t-progress__icon"
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12 23a11 11 0 100-22 11 11 0 000 22zM8.82 7.4L12 10.6l3.18-3.19 1.42 1.42L13.4 12l3.19 3.18-1.42 1.42L12 13.4 8.82 16.6 7.4 15.18 10.6 12 7.4 8.82 8.82 7.4z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :status > :status is success 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--success"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      <svg
        class="t-icon t-icon-check-circle-filled t-progress__icon"
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12 23a11 11 0 100-22 11 11 0 000 22zM7.5 10.59l3 3 6-6L17.91 9l-7.41 7.41L6.09 12l1.41-1.41z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :status > :status is success and percentage is 100 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--success"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 100%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      <svg
        class="t-icon t-icon-check-circle-filled t-progress__icon"
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12 23a11 11 0 100-22 11 11 0 000 22zM7.5 10.59l3 3 6-6L17.91 9l-7.41 7.41L6.09 12l1.41-1.41z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :status > :status is warning 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--warning"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      <svg
        class="t-icon t-icon-error-circle-filled t-progress__icon"
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12 1a11 11 0 110 22 11 11 0 010-22zm-1 13h2V6.5h-2V14zm2 1.5h-2v2h2v-2z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :status > :status is warning and percentage is 100 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--warning"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 100%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      <svg
        class="t-icon t-icon-error-circle-filled t-progress__icon"
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12 1a11 11 0 110 22 11 11 0 010-22zm-1 13h2V6.5h-2V14zm2 1.5h-2v2h2v-2z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :strokeWidth > :strokeWidth is Number, equal to 50 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="height: 50px; border-radius: 50px;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :strokeWidth > :strokeWidth is String, equal to "50px" 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="height: 50px; border-radius: 50px;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :theme > :theme circle progress has t-progress__circle-outer and t-progress__circle-inner class 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--circle t-progress--status--undefined"
    style="width: 112px; height: 112px; font-size: 20px;"
  >
    <div
      class="t-progress__info"
    >
      10%
    </div>
    <svg
      height="112"
      viewBox="0 0 112 112"
      width="112"
    >
      <circle
        class="t-progress__circle-outer"
        cx="56"
        cy="56"
        fill="none"
        r="53"
        stroke=""
        stroke-width="6"
      />
      <circle
        class="t-progress__circle-inner"
        cx="56"
        cy="56"
        fill="none"
        r="53"
        stroke-dasharray="33.30088212805181  299.7079391524663"
        stroke-linecap="round"
        stroke-width="6"
        transform="matrix(0,-1,1,0,0,112)"
      />
    </svg>
  </div>
</div>
`;

exports[`Progress > :props > :theme > :theme line and plump progress have t-progress__inner class 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :theme > :theme line and plump progress have t-progress__inner class 2`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress__bar t-progress--plump t-progress--under-ten"
  >
    <div
      class="t-progress__inner"
      style="width: 0%;"
    />
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :theme > :theme line and plump progress have t-progress__inner class 3`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress__bar t-progress--plump t-progress--under-ten"
  >
    <div
      class="t-progress__inner"
      style="width: 10%;"
    />
    <div
      class="t-progress__info"
    >
      10%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :theme > :theme line and plump progress have t-progress__inner class 4`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress__bar t-progress--plump t-progress--over-ten"
  >
    <div
      class="t-progress__inner"
      style="width: 11%;"
    >
      <div
        class="t-progress__info"
      >
        11%
      </div>
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :theme > :theme:circle 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--circle t-progress--status--undefined"
    style="width: 112px; height: 112px; font-size: 20px;"
  >
    <div
      class="t-progress__info"
    >
      0%
    </div>
    <svg
      height="112"
      viewBox="0 0 112 112"
      width="112"
    >
      <circle
        class="t-progress__circle-outer"
        cx="56"
        cy="56"
        fill="none"
        r="53"
        stroke=""
        stroke-width="6"
      />
    </svg>
  </div>
</div>
`;

exports[`Progress > :props > :theme > :theme:line 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :theme > :theme:plump 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress__bar t-progress--plump t-progress--under-ten"
  >
    <div
      class="t-progress__inner"
      style="width: 0%;"
    />
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > :props > :trackColor > :trackColor is "#aaa" 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="background-color: rgb(170, 170, 170); border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;

exports[`Progress > pure progress contains one class 1`] = `
<div
  class="t-progress"
>
  <div
    class="t-progress--thin t-progress--status--undefined"
  >
    <div
      class="t-progress__bar"
      style="border-radius: undefinedpx;"
    >
      <div
        class="t-progress__inner"
        style="width: 0%;"
      />
    </div>
    <div
      class="t-progress__info"
    >
      0%
    </div>
  </div>
</div>
`;
